apis/sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/write/WriteDocumentsService.java

```diff
@@ -332,6 +335,87 @@ public Uni<DocumentResponseWrapper<Void>> patchDocument(
         context);
   }
 
+  /**
+   * Sets data on a document with given ID in the given namespace and collection. Data will not get
+   * overwritten, allowing for partial updates of arrays and objects at various paths, all at once.
+   *
+   * @param table a CqlTable to be used for schema/validity checks
+   * @param namespace Namespace
+   * @param collection Collection name
+   * @param documentId The ID of the document to patch
+   * @param payload Document represented as JSON node, with all partial updates
+   * @param ttlAuto Whether to automatically determine TTL from the surrounding document
+   * @param context Execution content
+   * @return Document response wrapper containing the generated ID.
+   */
+  public Uni<DocumentResponseWrapper<Void>> setPathsOnDocument(
+      Uni<Schema.CqlTable> table,
+      String namespace,
+      String collection,
+      String documentId,
+      List<String> subPath,
+      JsonNode payload,
+      boolean ttlAuto,
+      ExecutionContext context) {
+    Uni<Integer> ttlValue = Uni.createFrom().item(0);
+    if (ttlAuto) {
+      ttlValue = determineTtl(namespace, collection, documentId, context);
+    }
+    return ttlValue
+        .onItem()
+        .transformToUni(
+            ttl ->
+                setPathsOnDocumentInternal(
+                    table, namespace, collection, documentId, subPath, payload, ttl, context));
+  }
+
+  private Set<List<String>> gatherAllPaths(
+      Iterator<String> fieldsIterator, List<String> subPathProcessed) {
+    Set<List<String>> allPaths = new HashSet<>();
+    while (fieldsIterator.hasNext()) {
+      String dottedField = fieldsIterator.next();
+      List<String> fieldPaths = processSubDocumentPath(Arrays.asList(dottedField.split("\\.")));
+      List<String> fullPath = new ArrayList<>();
+      fullPath.addAll(subPathProcessed);
+      fullPath.addAll(fieldPaths);
+      allPaths.add(fullPath);
+    }
+    return allPaths;
+  }
+
+  private Uni<DocumentResponseWrapper<Void>> setPathsOnDocumentInternal(
+      Uni<Schema.CqlTable> table,
+      String namespace,
+      String collection,
+      String documentId,
+      List<String> subPath,
+      JsonNode payload,
+      Integer ttl,
+      ExecutionContext context) {
+    // the payload will have a representation of all paths that need a partial update
+    return jsonSchemaManager
+        .validateJsonDocument(table, payload, true)
```

Comment:
This is technically just checking that there is no schema assigned, since the boolean here is for `has sub path`. It will error out if the user has any sort of schema and tries to $set.

One day we'll support schema on subdocument changes...

Reply:
Ahhh smart..